common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java

```diff
@@ -0,0 +1,113 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * A utility which collects any event's information given by calling markCode on such event(s).
+ * These events can be retrieved in csv format by calling method getFileContent.
+ * The event is recognized by a string marker which is prefixed with scenario code.
+ * MAX_SIZE_CODE_MARKER is the maximum number of markers this utility can have.
+ */
+public class CodeMarkerManager {
+
+    private boolean enableCodeMarker = false;
+    private int MAX_SIZE_CODE_MARKER = 1000;
+    private volatile List<CodeMarker> codeMarkers = new ArrayList<CodeMarker>();
+
+    //baseMilliSeconds is the time in milliseconds when first codemarker was captured.
+    private long baseMilliSeconds = 0;
+    private String scenarioCode = null;
+    private static CodeMarkerManager instance = null;
+
+    public static CodeMarkerManager getInstance() {
+        if(CodeMarkerManager.instance == null) {
```

Comment:
I don't think this check is needed twice i.e. you are already doing it inside the `synchronized` block...I don't think you need to do it here

Reply:
I have used "Double check" here. 
The double checked pattern is used to avoid obtaining the lock every time the code is executed.

See more here: https://www.baeldung.com/java-singleton-double-checked-locking
Let me know your thoughts. Thanks